docs/REST/authentication.md

```diff
@@ -461,3 +461,72 @@ def sign_request(host,
         "Authorization": "HMAC-SHA256 Credential=" + credential + "&SignedHeaders=" + signed_headers + "&Signature=" + signature
     }
 ```
+### PowerShell
+```PowerShell
+function Compute-SHA256Hash(
+    [string] $content
+)
+{
+    $sha256 = [System.Security.Cryptography.SHA256]::Create()
+    try {
+        return [Convert]::ToBase64String($sha256.ComputeHash([Text.Encoding]::ASCII.GetBytes($content)))
+    }
+    finally {
+        $sha256.Dispose()
+    }
+}
+
+function Compute-HMACSHA256Hash(
+    [string] $secret,      # base64 encoded
+    [string] $content
+)
+{
+    $hmac = [System.Security.Cryptography.HMACSHA256]::new([Convert]::FromBase64String($secret))
+    try {
+        return [Convert]::ToBase64String($hmac.ComputeHash([Text.Encoding]::ASCII.GetBytes($content)))
+    }
+    finally {
+        $hmac.Dispose()
+    }
+}
+
+function Sign-Request(
```

Comment:
I think we should make this look more explicit? 
Also, if possible, place it on the top of the snippet. It's about signing request to begin with.

```sh
Sign-Request(host, 
                      method,      // GET, PUT, POST, DELETE
                     url,         // path+query
                     body,        // request body (undefined of none)
                     credential,  // access key id
                     secret)      // access key value (base64 encoded)
```sh

Reply:
Sure, I moved Sign-Request to the top.  Could you clarify what you mean by making this look more explicit?  Is it about having separate paramters for host and url (path+query)?  I had seen that the javascript/python samples had separate parameters while the C#/Java/Golang didn't (host/path/query pulled out of the uri/url in the Request object).  I had gone with the Uri parameter because it is also the input into Invoke-RestMethod.

Reply:
This is a document that indeed focus on HMAC authentication.

The PS code snippet is to demonstrate how the HMAC is computed. It's not SDK or complete solution for PS scenarios. Therefore I think we should focus on the details and primitives outlined by the document. 

C#/Java/Golang snippets all work based on their respective HttpRequest objects, not on raw Uri.

Reply:
Ok, I updated Sign-Request to accept the host and url (path/query) like the javascript and python samples.

Note:  I added a qualifier to the parameter name `host` due to a name conflict with the PowerShell automatic variable also named `host`.